communication/mavsdkvehicleserver.cpp

```diff
@@ -296,6 +299,49 @@ MavsdkVehicleServer::MavsdkVehicleServer(QSharedPointer<VehicleState> vehicleSta
     mMavsdk->subscribe_on_new_system([this](){
         mMavlinkPassthrough.reset(new mavsdk::MavlinkPassthrough(mMavsdk->systems().at(0)));
 
+        mMavlinkPassthrough->subscribe_message(MAVLINK_MSG_ID_COMMAND_INT, [this](const mavlink_message_t &message) {
+
+            if(mavlink_msg_command_int_get_command(&message) == MAV_CMD_DO_SET_SERVO) {
+
+                mavlink_command_int_t cmd;
+                mavlink_msg_command_int_decode(&message, &cmd);
+
+                /*
+                 * Shuttle prototype GPIO configuration
+                 *  Pin 22: High/low gear
+                 *  Pin 27: T-lock servo (back)
+                */
+                unsigned int line_num = static_cast<int>(cmd.param1);
+                int gear_pwm = static_cast<int>(cmd.param2);
+
+                if(gear_pwm < 1000 || gear_pwm > 2000) {
+                    mavResult(MAV_CMD_DO_SET_SERVO, MAV_RESULT_DENIED, MAV_COMP_ID_AUTOPILOT1);
+                } else {
+                    gpiod::chip chip;
+                    QString chipName = "gpiochip4"; // Raspberry Pi 5
+                    gpiod::line gearSwitch;
+
+                    try {
+                        chip.open(chipName.toStdString(), 3);
+                        gearSwitch = chip.get_line(line_num);
+                        gearSwitch.request({"servo", gpiod::line_request::DIRECTION_OUTPUT, 0});
+
+                    } catch (const std::exception &e) {
+                        qDebug() << "Error: " << e.what();
+                    }
+
+                    for(int i = 0; i < 50; ++i) {   // 50 pulses for 1 second (20ms period)
+                        gearSwitch.set_value(1);
+                        usleep(gear_pwm);
+                        gearSwitch.set_value(0);
+                        usleep(20000 - gear_pwm);
+                        mavResult(MAV_CMD_DO_SET_SERVO, MAV_RESULT_ACCEPTED, MAV_COMP_ID_AUTOPILOT1);
+                    }
+                    gearSwitch.release();
```

Comment:
This looks good, but I would like to have it in a dedicated class. MavsdkVehicleServer should work in general for any vehicle and this code is very specific to the Shuttle.
A solution could be to create a new abstract class GearBox and a ServoGearBox could inherit from it and contain this code. This gearbox would contain the configuration of which GPIO to use and the mapping of gear to PWM. Then, the MovementController class gets a setGearBox and setGear which can be called from MavsdkVehicleServer (and itself calls the right GearBox implementation).
This would solve another issue I did not mention yet: MovementController->setSpeedToRPMFactor(..); (the factor for calculating m/s from rpm) needs to be set for each gear and could thus be handled within MovementController.